src/utils/hooks/withConfirmation/useChangeInputWithConfirmation.ts

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+
+import NiceModal from '@ebay/nice-modal-react';
+
+import {useTypedSelector} from '..';
+import {CONFIRMATION_DIALOG} from '../../../components/ConfirmationDialog/ConfirmationDialog';
+import {selectUserInput} from '../../../store/reducers/executeQuery';
+
+import i18n from './i18n';
+
+export async function getConfirmation(): Promise<boolean> {
+    return await NiceModal.show(CONFIRMATION_DIALOG, {
+        id: CONFIRMATION_DIALOG,
+        caption: i18n('context_unsaved-changes-warning'),
+        textButtonApply: i18n('action_apply'),
+    });
+}
+
+export function changeInputWithConfirmation<T>(callback: (args: T) => void) {
```

Comment:
I suppose thats not only about input, but about any callback action

useActionWithConfirmation ?